app/controllers/project.py

```diff
@@ -863,3 +872,98 @@ def post(self, project_id):
             status='success',
             message=f'project {project_id} enabled successfully'
         ), 201
+
+
+class ProjectPinView(Resource):
+    @jwt_required
+    def post(self , project_id):
+
+        current_user_id = get_jwt_identity()
+
+        project = Project.get_by_id(project_id)
+        user = User.get_by_id(current_user_id)
```

Comment:
These checks may not be necessary, please use the existing logic we have been using for validation
```suggestion
       if not project:
            return dict(
                status='fail',
                message=f'project {project_id} not found'
            ), 404

        if not is_owner_or_admin(project, current_user_id, current_user_roles):
            if not is_authorised_project_user(project, current_user_id, 'member'):
                return dict(status='fail', message='unauthorised'), 403
        
        if (project.pinned):
            return dict(
                status = 'fail',
                message = 'The project was already pinned'
            ),409
```